flask_multipass_saml_groups/provider.py

```diff
@@ -188,5 +189,11 @@ def _invalidate_session() -> Optional[Response]:
         if expires and expires < datetime.now(timezone.utc):
             session.clear()
 
+            next_url = request.url
+            url_info = urlsplit(next_url)
+            url_is_valid = not url_info.netloc or url_info.netloc == request.host
+            if next_url and url_is_valid:
+                session["_multipass_next_url"] = next_url
+
             return redirect(url_for(current_app.config["MULTIPASS_LOGIN_ENDPOINT"]))
```

Comment:
Would it not be sufficient to specify the `next_url` as a get param to the redirect url, as it seems that flask multipass already has the next_url logic implemented https://github.com/indico/flask-multipass/blob/8bf12747f4b15254032e8c7b30feee11d058c434/flask_multipass/core.py#L122-L126 ?

This might be a bit safer in the long run, assuming that a `next` query parameter would be more stable than an internal multipass mechanism. It might also avoid errors due to session resets, which may be done by flask multipass in the future.

Reply:
Thanks, indeed that should be the way to go since parsing and validation is done by `flask-multipass` later down the line. I updated the code to reflect this.